src/utils/getSymbolScale.ts

```diff
@@ -0,0 +1,31 @@
+import {Table, Scale} from '../types'
+import {assert} from './assert'
+import {GROUP_COL_KEY} from '../constants'
+import {scaleOrdinal} from 'd3-scale'
+import {SymbolType, symbols, symbolCircle} from 'd3-shape'
+
+// TODO pass in the symbols you'd like to use, as in color array.
+export const getSymbolScale = (
+  table: Table,
+  symbolColKeys: string[]
+): Scale<string, SymbolType> => {
+  if (!symbolColKeys.length) {
+    return (_i: string) => symbolCircle
+  }
+
+  const groupKeyCol = table.columns[GROUP_COL_KEY]
```

Comment:
Hm it seems like this is using the group column computed in the stat, except that group column is based off of the `fill` mapping instead of the `symbol` mapping.

So I think if the layer was specified like:
```
{
  type: 'scatterplot',
  fill: ['a', 'b'],
  symbol: ['c'],
  // ...
}
```
Then the symbol scale would be based off of the `a` and `b` columns rather than the `c` column, which is what was specified.

We can totally address this in a future PR, which might have been what you were thinking. I think the solution would be to append another group column to the table—so there would be one group column based off of the `fill` columns, and an independent column based off of the `symbol` columns.

Reply:
I am a little fuzzy on how we manipulate/interpret the columns, if you think it's ok I'll merge this now, and make an issue for myself to understand this properly.